contrib/rabbitmq/pkg/reconciler/resources/receive_adapter.go

```diff
@@ -0,0 +1,153 @@
+	"k8s.io/api/apps/v1"
+	"strconv"
+)
+
+type ReceiveAdapterArgs struct {
+	Image   string
+	Source  *v1alpha1.RabbitmqSource
+	Labels  map[string]string
+	SinkURI string
+}
+
+func MakeReceiveAdapter(args *ReceiveAdapterArgs) *v1.Deployment {
+	replicas := int32(1)
+
+	env := []corev1.EnvVar{
+		{
+			Name:  "RABBITMQ_BROKERS",
+			Value: args.Source.Spec.Brokers,
+		},
+		{
+			Name:  "RABBITMQ_TOPIC",
+			Value: args.Source.Spec.Topic,
+		},
+		{
+			Name: "RABBITMQ_USER",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: args.Source.Spec.User.SecretKeyRef,
+			},
+		},
+		{
+			Name: "RABBITMQ_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: args.Source.Spec.Password.SecretKeyRef,
+			},
+		},
+		{
+			Name:  "RABBITMQ_ROUTING_KEY",
+			Value: args.Source.Spec.QueueConfig.RoutingKey,
+		},
+		{
+			Name:  "RABBITMQ_CHANNEL_CONFIG_PREFETCH_COUNT",
+			Value: strconv.Itoa(args.Source.Spec.ChannelConfig.PrefetchCount),
+		},
+		{
+			Name:  "RABBITMQ_CHANNEL_CONFIG_QOS_GLOBAL",
+			Value: strconv.FormatBool(args.Source.Spec.ChannelConfig.GlobalQos),
+		},
+		{
+			Name:  "RABBITMQ_EXCHANGE_CONFIG_NAME",
+			Value: args.Source.Spec.ExchangeConfig.Name,
+		},
+		{
+			Name:  "RABBITMQ_EXCHANGE_CONFIG_TYPE",
+			Value: args.Source.Spec.ExchangeConfig.TypeOf,
+		},
+		{
+			Name: "RABBITMQ_EXCHANGE_CONFIG_DURABLE",
+			Value: strconv.FormatBool(args.Source.Spec.ExchangeConfig.Durable),
+		},
+		{
+			Name: "RABBITMQ_EXCHANGE_CONFIG_AUTO_DELETED",
+			Value: strconv.FormatBool(args.Source.Spec.ExchangeConfig.AutoDeleted),
+		},
+		{
+			Name: "RABBITMQ_EXCHANGE_CONFIG_INTERNAL",
+			Value: strconv.FormatBool(args.Source.Spec.ExchangeConfig.Internal),
+		},
+		{
+			Name: "RABBITMQ_EXCHANGE_CONFIG_NOWAIT",
+			Value: strconv.FormatBool(args.Source.Spec.ExchangeConfig.NoWait),
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_NAME",
+			Value: args.Source.Spec.QueueConfig.Name,
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_DURABLE",
+			Value: strconv.FormatBool(args.Source.Spec.QueueConfig.Durable),
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_AUTO_DELETED",
+			Value: strconv.FormatBool(args.Source.Spec.QueueConfig.DeleteWhenUnused),
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_EXCLUSIVE",
+			Value: strconv.FormatBool(args.Source.Spec.QueueConfig.Exclusive),
+		},
+		{
+			Name: "RABBITMQ_QUEUE_CONFIG_NOWAIT",
+			Value: strconv.FormatBool(args.Source.Spec.QueueConfig.NoWait),
+		},
+		{
+			Name: "SINK_URI",
+			Value: args.SinkURI,
+		},
+	}
+
+	return &v1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    args.Source.Namespace,
+			GenerateName: fmt.Sprintf("%s-", args.Source.Name),
+			Labels:       args.Labels,
+		},
+		Spec: v1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: args.Labels,
+			},
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"sidecar.istio.io/inject": "true",
```

Comment:
Knative eventing no longer requires Istio. Are you sure you want to force this on?